packages/app-project/src/screens/ProjectHomePage/components/ZooniverseTalk/components/RecentSubjects/helpers/fetchRecentSubjects/fetchSubjectData.js

```diff
@@ -1,11 +1,26 @@
 import { panoptes } from '@zooniverse/panoptes-js'
+import logToSentry from '@helpers/logger/logToSentry.js'
 
-async function fetchSubjectData (subjectIds) {
-  const response = await panoptes.get('/subjects', {
-    id: subjectIds.join(',')
-  })
-  const { subjects } = response.body
-  return subjectIds.map(subjectID => subjects.find(subject => subject.id === subjectID))
+async function fetchSubjectData(subjectIds) {
+  let recentlyDiscussedSubjects = []
+  try {
+    const response = await panoptes.get('/subjects', {
+      id: subjectIds.join(',')
+    })
+    const { subjects } = response.body
+    // response does not come back in the same order as subjectIds array
+    // so we have to order them correctly here
+    recentlyDiscussedSubjects = subjectIds.reduce((acc, id) => {
+      // sometimes old subjects have been deleted, but their Talk thread still exists via their subject.id
+      if (subjects.find(subject => subject.id === id)) {
+        acc.push(subjects.find(subject => subject.id === id))
+      }
```

Comment:
Optionally, could do the `subjects.find` once and reuse result:
```suggestion
      const subject = subjects.find(subject => subject.id === id);
      if (subject) {
        acc.push(subject);
      }
```

Reply:
Good point!